modules/planning/open_space/coarse_trajectory_generator/hybrid_a_star.cc

```diff
@@ -411,10 +394,10 @@ bool HybridAStar::GenerateSCurveSpeedAcceleration(HybridAStartResult* result) {
   const size_t num_of_knots = x_size - 1;
 
   PiecewiseJerkSpeedProblem path_time_qp(num_of_knots, delta_t_, init_s);
-  path_time_qp.set_weight_x(w[0]);
-  path_time_qp.set_weight_dx(w[1]);
-  path_time_qp.set_weight_ddx(w[2]);
-  path_time_qp.set_weight_dddx(w[3]);
+  path_time_qp.set_weight_ddx(planner_open_space_config_.warm_start_config()
+                              .s_curve_config().acc_weight());
+  path_time_qp.set_weight_dddx(planner_open_space_config_.warm_start_config()
+                               .s_curve_config().jerk_weight());
```

Comment:
DRY: Save `planner_open_space_config_.warm_start_config().s_curve_config()` to var and reuse.